crates/config/src/providers/remappings.rs

```diff
@@ -246,50 +244,67 @@ impl RemappingsProvider<'_> {
-            .inspect(|lib| {
-                trace!("find all remappings of nested foundry.toml lib: {:?}", lib);
+            .flat_map_iter(|lib| {
+                trace!(?lib, "find all remappings of nested foundry.toml");
+                self.nested_foundry_remappings(&lib)
             })
-            .flat_map(|lib: PathBuf| {
-                // load config, of the nested lib if it exists
-                let Ok(config) = Config::load_with_root(&lib) else { return vec![] };
-                let config = config.sanitized();
-
-                // if the configured _src_ directory is set to something that
-                // [Remapping::find_many()] doesn't classify as a src directory (src, contracts,
-                // lib), then we need to manually add a remapping here
-                let mut src_remapping = None;
-                if ![Path::new("src"), Path::new("contracts"), Path::new("lib")]
-                    .contains(&config.src.as_path())
-                    && let Some(name) = lib.file_name().and_then(|s| s.to_str())
-                {
-                    let mut r = Remapping {
-                        context: None,
-                        name: format!("{name}/"),
-                        path: format!("{}", lib.join(&config.src).display()),
-                    };
-                    if !r.path.ends_with('/') {
-                        r.path.push('/')
-                    }
-                    src_remapping = Some(r);
-                }
+            .collect::<Vec<_>>()
+            .into_iter()
+    }
 
-                // Eventually, we could set context for remappings at this location,
-                // taking into account the OS platform. We'll need to be able to handle nested
-                // contexts depending on dependencies for this to work.
-                // For now, we just leave the default context (none).
-                let mut remappings =
-                    config.remappings.into_iter().map(Remapping::from).collect::<Vec<Remapping>>();
+    fn nested_foundry_remappings(&self, lib: &Path) -> Vec<Remapping> {
+        // load config, of the nested lib if it exists
+        let Ok(config) = Config::load_with_root(lib) else { return vec![] };
+        let config = config.sanitized();
+
+        // if the configured _src_ directory is set to something that
+        // `Remapping::find_many` doesn't classify as a src directory (src, contracts,
+        // lib), then we need to manually add a remapping here
+        let mut src_remapping = None;
+        if ![Path::new("src"), Path::new("contracts"), Path::new("lib")]
+            .contains(&config.src.as_path())
+            && let Some(name) = lib.file_name().and_then(|s| s.to_str())
+        {
+            let mut r = Remapping {
+                context: None,
+                name: format!("{name}/"),
+                path: format!("{}", lib.join(&config.src).display()),
+            };
+            if !r.path.ends_with('/') {
```

Comment:
not this PR related but wonder if we should apply here same check as in https://github.com/foundry-rs/compilers/commit/f6e4e6a441f6ff7d45c8f0ddcf8b786252433c41 that is if nested libs has single files remapping

Reply:
nm, were here looking remappings for deps / dirs, so doesn't apply, tested with project from https://github.com/foundry-rs/foundry/issues/6706#issuecomment-3141270852 as a dep and remappings looks good